### ISSUE TITLE
feat: coverage diff change

### ISSUE THREAD

**Issue by vladopajic** [vladopajic/go-test-coverage#42]
some developers prefer to see coverage difference change on their PRs. `go-test-coverage` could be extended to allow such functionality for those who want it.

1) give new option which will show error if coverage difference from base branch is below threshold.
`diff-threshold` property will set allowed threshold for coverage difference (in percentage) from base branch. for example, if this property is set to to 5 (meaning 5%), error will be shown if current PR doesn't have at least 5% more code covered. similar, if value is set to -5, PR will be allowed to have up to 5% less covered code then base branch. 
this property will stack with other rules.

2) give new option to post comment on PR with coverage difference.
`diff-comments` property will control if GitHub action will post comments on PR with coverage difference.

**Comment by vxjohanforssell** [comment]
I'm currently running the following little script to generate coverage on changed packages only: 

```
MODULE=$(head -1 go.mod | awk '{print $2}')
PACKAGES=$(git diff --name-only --no-renames origin/master HEAD | grep '\.go$' | sed 's|/.*||' | uniq)
TESTABLE_PACKAGES=""
for P in $PACKAGES; do
  TESTABLE_PACKAGES="$TESTABLE_PACKAGES $MODULE/$P/..."
done
go test -coverprofile=coverage.out -covermode=atomic $TESTABLE_PACKAGES
```

**Comment by vladopajic** [comment]
closing this issue as it was implemented in `v2.11.0`, here is the  [documentation](https://github.com/vladopajic/go-test-coverage/blob/main/docs/github_action.md#report-coverage-difference) and  [PR](https://github.com/vladopajic/go-test-coverage/pull/129) with example report.

---

@zhongyi-zhang, @vxjohanforssell, @andriisoldatenko, @prashantv, @gemyago, hello all, i hope this comment finds you well! i am kindly notifying you about progress of this feature. happy coding 🌞

**Comment by andriisoldatenko** [comment]
@vladopajic thanks a lot for your effort! I've tried example from docs and it works, it prints comment with code coverage report.

I have a few questions regarding:

`diff-comments` and `go-test-coverage` I can't find in [docs](https://github.com/vladopajic/go-test-coverage/blob/main/action.yml) or just by searching inside source code where to set up these 2 options.

**Comment by vladopajic** [comment]
hey @andriisoldatenko, 

actually `diff-comments` wasn't really used, and instead `breakdown-file-name` and `diff-base-breakdown-file-name` inputs control most of diff 

please see [test.yml](https://github.com/vladopajic/go-test-coverage/blob/main/.github/workflows/test.yml) of this repo to see how coverage diff is used. see also [docs](https://github.com/vladopajic/go-test-coverage/blob/main/docs/github_action.md#report-coverage-difference).

**Comment by andriisoldatenko** [comment]
> hey [@andriisoldatenko](https://github.com/andriisoldatenko),
> 
> actually `diff-comments` wasn't really used, and instead `breakdown-file-name` and `diff-base-breakdown-file-name` inputs control most of diff feature.
> 
> please see [test.yml](https://github.com/vladopajic/go-test-coverage/blob/main/.github/workflows/test.yml) of this repo to see how coverage diff is used. see also [docs](https://github.com/vladopajic/go-test-coverage/blob/main/docs/github_action.md#report-coverage-difference).

yeah that's what I noticed! cool thanks

**Comment by vladopajic** [comment]
for now, you can't control `diff-threshold` (it's not implemented). but likely it will be implemented soon (it shouldn't be hard to implement it)

**Comment by vladopajic** [comment]
added #181 issue to track progress for `diff-threshold` and #182 to track progress of posting comments to PR.

initially i envisioned both to be part of this feature, but since this feature was already large i decided to merge it as just "report only". and added docs where other github action was used to post comments to PR.

#181 (diff-threshold) will likely get implemented in near future, while #182 i am not really sure when since this just doesn't add any necessary functionality.
